homeassistant/components/influxdb/__init__.py

```diff
@@ -161,6 +163,9 @@ def validate_version_specific_config(conf: Dict) -> Dict:
         ),
         vol.Optional(CONF_RETRY_COUNT, default=0): cv.positive_int,
         vol.Optional(CONF_DEFAULT_MEASUREMENT): cv.string,
+        vol.Optional(CONF_MEASUREMENT_ATTR, default=DEFAULT_MEASUREMENT_ATTR): vol.In(
+            ["unit_of_measurement", "domain", "entity_id"]
```

Comment:
We should not add `domain`. Instead we should an option to do by domain and device class.

Reply:
So you're thinking of something like measurement: `domain`_`device_class`?

`binary_sensor_door`
`sensor_humidity`
`cover_shutter`

And, in the (many) cases where device class isn't set, just the domain?

Reply:
yes. And let's do `__` to separate domain and device class.

Reply:
This is implemented in 7a57484